src/components/AnnouncementsModal.js

```diff
@@ -0,0 +1,305 @@
+}) => {
+  return (
+    <ListItem
+      padding={4}
+      spacing={3}
+      css={`
+        cursor: pointer;
+        border-bottom: ${getValueForAnnouncementItem('borderColor')} 1px solid;
+        background-color: ${selected
+          ? getValueForAnnouncementItem('selected.backgroundColor')
+          : 'inherit'};
+
+        :hover {
+          background-color: ${selected
+            ? getValueForAnnouncementItem('selected.hoverBackgroundColor')
+            : getValueForAnnouncementItem('hoverBackgroundColor')};
+        }
+      `}
+      onClick={() => {
+        setSelectedAnnouncementId(id);
+      }}
+    >
+      {seen ? <Icon name={Icons.EYE_SLASH} /> : <Icon name={Icons.EYE} />}
+      <Box
+        as="span"
+        css={`
+          white-space: nowrap;
+          overflow: hidden;
+          text-overflow: ellipsis;
+        `}
+      >
+        {title}
+      </Box>
+    </ListItem>
+  );
+};
+
+AnnouncementItem.propTypes = {
+  id: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  selected: PropTypes.bool,
+  seen: PropTypes.bool,
+  setSelectedAnnouncementId: PropTypes.func,
+};
+
+AnnouncementItem.defaultProps = {
+  selected: false,
+  seen: false,
+};
+
+const AnnouncementsList = ({
+  announcements,
+  selectedAnnouncementId,
+  seenAnnouncements,
+  setSelectedAnnouncementId,
+}) => (
+  <List
+    css={`
+      width: 30%;
+      border-right: ${getValueForAnnouncementList('borderColor')} 1px solid;
+      overflow-y: auto;
+    `}
+  >
+    {announcements.map((announcement) => {
+      return (
+        <AnnouncementItem
+          key={announcement.uid}
+          id={announcement.uid}
+          title={announcement.title}
+          selected={announcement.uid === selectedAnnouncementId}
+          seen={seenAnnouncements.includes(announcement.uid)}
+          setSelectedAnnouncementId={setSelectedAnnouncementId}
+        />
+      );
+    })}
+  </List>
+);
+
+AnnouncementsList.propTypes = {
+  announcements: PropTypes.array,
+  selectedAnnouncementId: PropTypes.string,
+  seenAnnouncements: PropTypes.array,
+  setSelectedAnnouncementId: PropTypes.func,
+};
+
+AnnouncementsList.defaultProps = {
+  seenAnnouncements: [],
+};
+
+const AnnouncementContent = ({
+  title,
+  imageSrc,
+  body,
+  callToAction,
+  callToActionLabel,
+}) => (
+  <Stack
+    forwardedAs="article"
+    padding={4}
+    spacing={3}
+    css={`
+      width: 70%;
+    `}
+  >
+    <Title>{title}</Title>
+    {imageSrc && (
+      <Link href={callToAction}>
+        <Image
+          src={imageSrc}
+          alt={callToActionLabel}
+          css={`
+            width: 100%;
+            max-height: 30vh;
+            background-position: center center;
+            background-repeat: no-repeat;
+            object-fit: cover;
+
+            &:hover {
+              opacity: 0.85;
+            }
+          `}
+        />
+      </Link>
+    )}
+
+    <div
+      dangerouslySetInnerHTML={{ __html: body }}
```

Comment:
Nit: Maybe we should investigate a way to render this HTML with some kind of whitelist, using e.g. https://github.com/cure53/DOMPurify ?
@geoffreydhuyvetters do you have experience with cases like this where we can't block all HTML from being rendered, but should also prevent a security issue in an external system from causing an XSS vulnerability in our app?

Reply:
To clarify: We're fetching this data from our own CMS for our public website, publiq.be, which in theory we can trust. But it's not impossible that that gets hacked at some point and someone inserts script tags etc into the content that we're fetching and rendering here.

Reply:
We'll make this a separate ticket and PR?

Reply:
👍